### PR TITLE
Basic IME mapping

### PR DESCRIPTION
**Pull request by EReeves**
Fixes https://github.com/mvlabat/bevy_egui/issues/145

This is enough for my needs, but FYI:
- `Ime::Disabled` just maps to `Event::CompositionEnd` with an empty string for text. You could capture `Ime::Preedit` as state to handle it that way. I don't know where best to add the state.

### CODE REVIEW

File: src/systems.rs [hunk 1]
```diff
@@ -231,6 +233,39 @@ pub fn process_input_system(
...
+            .egui_input
+            .events
+            .push(event);
+    }
+
+    for ev in input_events.ev_ime_input.iter() {
+        match ev {
+            Ime::Preedit {
+                window,
+                value: _,
```
Comment (mvlabat) [inline src/systems.rs:250]: shouldn't we also push a `CompositionUpdate` event with this value right after pushing `CompositionStart`?

(looking at the reference implementation at https://github.com/emilk/egui/blob/2bc2fb9c393bbbfc4832a35af1c100bc4ea48719/crates/egui-winit/src/lib.rs#L301)

**Comment by mvlabat** [comment]
Hey! Thank you for the PR and apologies for taking a while to review it. Looks good to me in general, left 1 comment about the `Ime::Preedit` event though

File: src/systems.rs (hunk 1 above)
Comment (EReeves) [reply src/systems.rs]: Sounds good to me. I read through the reference and all code that uses the event. The event is used by TextEdit to preview text before it's committed by the IME. It should also be added for the sake of completion.

**Review by liquidhelium — APPROVED** [review]

**Review by liquidhelium — APPROVED** [review]

File: src/systems.rs (hunk 1 above)
Comment (mvlabat) [reply src/systems.rs]: Sorry for forgetting to comment on this again, I feel really bad for neglecting this PR for several releases. The last question I want to ask is if there is a reason why we don't align our implementation to the reference one completely.

I haven't had a chance to test it myself and to understand if there's really any difference in behaviour, but I guess I'd feel safer if we just adopted exactly the same logic.

**Comment by v-kat** [comment]
Will this get merged soon?

**Comment by v-kat** [comment]
Is this related to WASM mobile keyboard display at all? I've been trying to debug it and had assumed it was an winit issue or maybe something with IME (everything works fine with just egui).

**Pull request merged**